src/webview/js/toggleSpoiler.js

```diff
@@ -0,0 +1,59 @@
+/* @flow strict-local */
+/* eslint-disable no-useless-return */
+
+const collapseSpoiler = (spoiler: HTMLElement) => {
+  const spoiler_height = spoiler.scrollHeight;
+
+  // Set height to rendered height on next frame, then to zero on following
+  // frame to allow CSS transition animation to work
+  requestAnimationFrame(() => {
+    spoiler.style.height = `${spoiler_height}px`;
+    spoiler.classList.remove('spoiler-content-open');
+
+    requestAnimationFrame(() => {
+      spoiler.style.height = '0px';
+    });
+  });
+};
```

Comment:
Huh, this is complicated. Is this just based on what the webapp does?

I see, yes, it is. That's an important thing to mention in a comment in the code, then, for a couple of reasons:
* For any questions the reader has about why the code seems odd, it provides a pointer for where to look: go ask about the webapp. :slightly_smiling_face: 
* Most importantly, when someone's considering *changing* the code, it tells them to probably stop -- one of the important things to maintain about this code will be that it should be a lot like the webapp's corresponding code.